x-pack/plugin/enrich/src/test/java/org/elasticsearch/xpack/enrich/EnrichPolicyExecutorTests.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.enrich;
+
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.function.LongSupplier;
+
+import org.elasticsearch.ElasticsearchException;
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.action.LatchedActionListener;
+import org.elasticsearch.client.Client;
+import org.elasticsearch.cluster.metadata.IndexNameExpressionResolver;
+import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.test.ESTestCase;
+import org.elasticsearch.threadpool.TestThreadPool;
+import org.elasticsearch.threadpool.ThreadPool;
+import org.elasticsearch.xpack.core.enrich.EnrichPolicy;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.hamcrest.CoreMatchers.instanceOf;
+
+public class EnrichPolicyExecutorTests extends ESTestCase {
+
+    private static ThreadPool testThreadPool;
+    private static ActionListener<PolicyExecutionResult> noOpListener = new ActionListener<>() {
+        @Override
+        public void onResponse(PolicyExecutionResult policyExecutionResult) { }
+
+        @Override
+        public void onFailure(Exception e) { }
+    };
+
+    @BeforeClass
+    public static void beforeCLass() {
+        testThreadPool = new TestThreadPool("EnrichPolicyExecutorTests");
+    }
+
+    @AfterClass
+    public static void afterClass() {
+        ThreadPool.terminate(testThreadPool, 30, TimeUnit.SECONDS);
+        testThreadPool = null;
```

Comment:
more curious ... is this needed to null this out ?

Reply:
I had to go searching for examples in the code base for creating test thread pools, and this was just the existing logic for tearing it down in the test I was in. I don't think it's really required, I'll remove it.